Libraries/ActionSheetIOS/RCTActionSheetManager.m

```diff
@@ -110,7 +113,32 @@ - (dispatch_queue_t)methodQueue
     };
   }
 
-  [ctrl presentViewController:share animated:YES completion:nil];
+  /*
+   * The `anchor` option takes a view to set as the anchor for the share
+   * popup to point to, on iPads running iOS 8. If it is not passed, it
+   * defaults to centering the share popup on screen without any arrows.
+   */
+  if ([share respondsToSelector:@selector(popoverPresentationController)]) {
+    NSNumber *anchorViewTag = [RCTConvert NSNumber:options[@"anchor"]];
+    if (anchorViewTag) {
+      dispatch_async(self.bridge.uiManager.methodQueue, ^{
+        [self.bridge.uiManager addUIBlock:^(__unused RCTUIManager *uiManager, RCTSparseArray *viewRegistry){
+          UIView *anchorView = viewRegistry[anchorViewTag];
+          share.popoverPresentationController.sourceView = ctrl.view;
+          share.popoverPresentationController.sourceRect = [anchorView convertRect:anchorView.bounds toView:ctrl.view];
+          [ctrl presentViewController:share animated:YES completion:nil];
```

Comment:
You can just extract the block definition to a local variable, dispatch_async in one case, and invoke the bloke yourself in the other case.